packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -99,3 +101,76 @@ void setExitFunctionForTests([ExitFunction exitFunction]) {
 void restoreExitFunction() {
   _exitFunction = _defaultExitFunction;
 }
+
+/// A portable version of [io.ProcessSignal].
+///
+/// Listening on signals that don't exist on the current platform is just a
+/// no-op. This is in contrast to [io.ProcessSignal], where listening to
+/// non-existent signals throws an exception.
+class ProcessSignal implements io.ProcessSignal {
+  static const ProcessSignal SIGWINCH = const _PosixProcessSignal._(io.ProcessSignal.SIGWINCH);
+  static const ProcessSignal SIGTERM = const _PosixProcessSignal._(io.ProcessSignal.SIGTERM);
+  static const ProcessSignal SIGUSR1 = const _PosixProcessSignal._(io.ProcessSignal.SIGUSR1);
+  static const ProcessSignal SIGUSR2 = const _PosixProcessSignal._(io.ProcessSignal.SIGUSR2);
+
+  static final ProcessSignal SIGINT = SigintProcessSignal.instance; // ignore: non_constant_identifier_names
+
+  final io.ProcessSignal _wrappedSignal;
+
+  const ProcessSignal._(this._wrappedSignal);
```

Comment:
constructors should go first, so that it's obvious there's no implicit constructor